docs/general/session.md

```diff
@@ -0,0 +1,22 @@
+# Semantic conventions for session
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions to apply to client-side applications when tracking sessions.
+
+Session is defined as the period of time encompassing all activities performed by the application and the actions
+executed by the end user.
+
+Consequently, a Session is represented as a collection of Logs, Events, and Spans emitted by the Client Application
+throughout the Session's duration. Each Session is assigned a unique identifier, which is included as an attribute in
+the Logs, Events, and Spans generated during the Session's lifecycle.
+
+## Attributes
+
+<!-- semconv session-id -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `session.id` | string | A UUID to identify the current session. This value should be generated by the SDK and not supplied by the client. | `00112233-4455-6677-8899-aabbccddeeff` | Opt-In |
```

Comment:
* Why "not supplied by the client"?
* Why does it have to be UUID format?
* How does the client exercise control over the life cycle of a session that reflect the business requirements of the app?

Reply:
The 'not supplied by the client' requires rewording. I guess @bryce-b meant the instrumentations should not add this attribute to the signals, but only a processor at SDK level should be adding it. I don't know if this fact needs to be mentioned here though.

Reply:
Regarding UUID, note that the session id has to be unique in the context of a specific resource. For mobile apps, we said the resource will have a `device.id`; For browser apps, we said we will have `browser.visitor.id` but it's not added to the semantic conventions yet. I think we tend to use UUIDs since it's easier to guarantee uniqueness, otherwise we will need to remember a state for that purpose, and this adds complexity on the client devices. Should we reword the text to say `unique id` in place of `UUID`?

Reply:
Regarding the lifecycle of a session, we want to omit specifying it here. This PR originally started describing it but there are many variations to it from different vendors. Since this id is added using a processor at the SDK level we said each vendor could have their own implementation managing the lifecycle of a session and only thing needed in the spec is how the session identifier is represented in the signals. Let me know if this makes sense.

Reply:
> Should we reword the text to say unique id in place of UUID?

Yes, the description should capture semantic requirements, not implementation details like whether it's UUID or not.

Reply:
> but only a processor at SDK level should be adding it

This again sounds like an implementation detail not relevant to the semantic meaning of `session.id`. If my instrumentation has its own ideas about what constitutes a session ID and what its life cycle is, why should semantic convention prevent me from setting it explicitly and instead force on relying some unspecified vendor implementation?

Reply:
Sounds good. 
@bryce-b how about we trim the text to "A unique id to identify a session".